sktime/transformations/panel/supervised_intervals.py

```diff
@@ -0,0 +1,471 @@
+            )
+
+        if self.min_interval_length < 3:
+            self._min_interval_length = 3
+
+        if self.series_length_ < 6:
+            raise ValueError("Series length must be at least 6.")
+        elif self._min_interval_length * 2 > self.series_length_:
+            raise ValueError(
+                "Minimum interval length must be less than half the series length."
+            )
+
+        if self.features is None:
+            self._features = [
+                row_mean,
+                row_median,
+                row_std,
+                row_slope,
+                row_numba_min,
+                row_numba_max,
+                row_iqr,
+                row_count_mean_crossing,
+                row_count_above_mean,
+            ]
+
+        if not isinstance(self._features, list):
+            self._features = [self._features]
+
+        self._threads_to_use = check_n_jobs(self.n_jobs)
+
+        le = preprocessing.LabelEncoder()
+        return le.fit_transform(y)
+
+    def _generate_intervals(self, X, X_norm, y, idx, keep_transform):
+        rs = 255 if self.random_state == 0 else self.random_state
+        rs = (
+            None
+            if self.random_state is None
+            else (rs * 37 * (idx + 1)) % np.iinfo(np.int32).max
+        )
+        rng = check_random_state(rs)
+
+        Xt = np.empty((self.n_instances_, 1)) if keep_transform else None
+        candidate_agg_feats = []
+
+        for i in range(self.n_dims_):
+            for feature in self._features:
+                random_cut_point = int(
+                    rng.randint(
+                        self._min_interval_length,
+                        self.series_length_ - self._min_interval_length,
+                    )
+                )
+
+                candidate_agg_feat_L, Xt_L = self._supervised_search(
+                    X_norm[:, i, :random_cut_point],
+                    y,
+                    0,
+                    feature,
+                    i,
+                    X[:, i, :],
+                    rng,
+                    keep_transform,
+                )
+                candidate_agg_feats.extend(candidate_agg_feat_L)
+                if keep_transform:
+                    Xt = np.hstack((Xt, Xt_L))
+
+                candidate_agg_feat_R, Xt_R = self._supervised_search(
+                    X_norm[:, i, random_cut_point:],
+                    y,
+                    random_cut_point,
+                    feature,
+                    i,
+                    X[:, i, :],
+                    rng,
+                    keep_transform,
+                )
+                candidate_agg_feats.extend(candidate_agg_feat_R)
+                if keep_transform:
+                    Xt = np.hstack((Xt, Xt_R))
+
+        if keep_transform:
+            Xt = Xt[:, 1:]
+
+        return candidate_agg_feats, Xt
+
+    def _transform_intervals(self, X, idx, skip):
```

Comment:
can this be turned into njit? seems to be a bottleneck?

Reply:
we dont know if the `feature` function stored in `self._intervals` will be numba compatible. Best leave it to the user to pass in efficient functions, the default ones are already all numba.